modules/bullet/space_bullet.cpp

```diff
@@ -457,24 +470,43 @@ void SpaceBullet::reload_collision_filters(AreaBullet *p_area) {
 	dynamicsWorld->refreshBroadphaseProxy(ghost_object);
 }
 
+void SpaceBullet::register_collision_object(CollisionObjectBullet *p_object) {
+	collision_objects.push_back(p_object);
+}
+
+void SpaceBullet::unregister_collision_object(CollisionObjectBullet *p_object) {
+	collision_objects.erase(p_object);
+}
+
 void SpaceBullet::add_rigid_body(RigidBodyBullet *p_body) {
+#ifdef TOOLS_ENABLED
+	// This can't never happen at this piont.
+	CRASH_COND(p_body->is_in_world);
+#endif
 	if (p_body->is_static()) {
```

Comment:
Again, I'm not convinced this will never happen. We should use `ERR_FAIL_COND_MSG` with a good explanation, instead of `CRASH_COND`.

Reply:
Same as above, I'm going to just enrich the comment.